### PR TITLE
Add liveness probes for all celery workers

### PR DESCRIPTION
**Pull request by bufke**
Based on django-helm-chart's version which itself is based on https://github.com/celery/celery/issues/4079

I found the syntax pretty finicky. The grep is required to make it return the correct exit code. Don't trust any one comment on that issue - everyone everywhere is wrong.

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -83,7 +83,16 @@ kpi:
...
-    resources:
+    livenessProbe:
+      enabled: true
+      initialDelaySeconds: 10
+      periodSeconds: 60
+      timeoutSeconds: 30
+      command:
+        - /bin/sh
+        - -c
+        - exec
```
Comment (noliveleger) [inline values.yaml:94]: To make it work, I had to remove `exec` from the command. 
With  `exec`, event `Unhealthy` seems to never be triggered.  

To test, I just stopped redis and wait for the pod to restart.
I decreased `periodSeconds` to 30 and `timeoutSeconds` to 10 to test. Pod should restart within the next two minutes Redis has been stopped.

File: values.yaml [hunk 2]
```diff
@@ -98,6 +107,16 @@ kpi:
...
     replicaCount: 1
+    livenessProbe:
+      enabled: true
+      initialDelaySeconds: 10
+      periodSeconds: 60
+      timeoutSeconds: 30
+      command:
+        - /bin/sh
+        - -c
+        - exec
```
Comment (noliveleger) [inline values.yaml:118]: I guess `exec` should be removed over here too ;-)

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: values.yaml [hunk 3]
```diff
@@ -188,6 +207,16 @@ kobocat:
...
```
Comment (noliveleger) [inline values.yaml:218]: https://github.com/kobotoolbox/kobo-helm-chart/pull/40#discussion_r1189189982

File: templates/kpi/deployment-worker.yaml [hunk 4]
```diff
@@ -31,6 +31,14 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.kpi.image.repository }}:{{ .Values.kpi.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.kpi.image.pullPolicy }}
+          {{- if .Values.kpi.worker.livenessProbe.enabled }}
```
[the remark this answers is not in the document]
Comment (bufke) [reply templates/kpi/deployment-worker.yaml:34]: I looked up how bitnami charts do this, but the answer is incredibly complex with tons of code. @Yann-J is there a specific use case where you'd want to override more?

File: values.yaml (hunk 1 above)
Comment (bufke) [reply values.yaml:94]: The only variation that I could get working was this

```
    command:
      - "bash"
      - "-c"
      - "celery -A glitchtip inspect ping -d celery@$HOSTNAME | grep -q OK"
```

I will go ahead and switch it to `{{- .Values.kpi.worker.livenessProbe | toYaml | nindent 12 }}` as @Yann-J suggested to cover those additional options. And make the change to the upstream django-helm-chart.

I tested by starting and stopping redis. I'm confused why the /bin/sh version doesn't appear to work for me.

File: templates/kpi/deployment-worker.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (bufke) [reply templates/kpi/deployment-worker.yaml:34]: Upstream change https://gitlab.com/burke-software/django-helm-chart/-/commit/3099e678fcf3c32e7218e6fcf9ecbfe3203a5411

File: values.yaml (hunk 1 above)
Comment (noliveleger) [reply values.yaml]: > `"celery -A glitchtip` 

Maybe workers are not using the same base image in GT and Kobo (I tested it with Kobo).

Comment (bufke) [reply values.yaml]: Let me double check - I would explicitly like to keep aligned to django-helm-chart which we use and glitchtip uses.

Comment (bufke) [reply values.yaml]: Wow you're right - I can't explain it though. Both projects use python:3.10-slim as the base image. I still think we should keep aligned to django-helm-chart and that it needs to use bash as it works in more observed cases. If this looks good as is, please merge.

**Review by noliveleger — APPROVED** [review]

**Pull request merged**